### PR TITLE
Added details about build output location.

### PR DESCRIPTION
**Pull request by niccottrell**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -12,6 +12,8 @@ the documentation locally if you have `giza
      cd docs/
      make html
 
+If successful, the built HTML files will be located in ``build/master/html/``.
```
Comment (i80and) [inline README.rst]: This isn't necessarily true; when we branch off stable releases, the build will happen under that branch's directory.

Comment (niccottrell) [reply README.rst]: Updated. Is this getting too messy now?

**Comment by rkumar-mongo** [comment]
@i80and looking at it, is this guide still true? there are no references to sphinx or mut here, as I'm looking through. I've got no objections to the fixups as they are otherwise.

**Comment by i80and** [comment]
@rkumar-mongo Our repository READMEs are all comically out of date and shouldn't be used. But this fixup is reasonable in isolation.

**Pull request merged**